crates/node/core/src/version.rs

```diff
@@ -82,9 +80,53 @@ pub fn default_extra_data_bytes() -> Bytes {
 /// The default client version accessing the database.
 pub fn default_client_version() -> ClientVersion {
     ClientVersion {
-        version: CARGO_PKG_VERSION.to_string(),
-        git_sha: VERGEN_GIT_SHA.to_string(),
-        build_timestamp: VERGEN_BUILD_TIMESTAMP.to_string(),
+        version: env!("CARGO_PKG_VERSION").to_string(),
+        git_sha: env!("VERGEN_GIT_SHA_SHORT").to_string(),
+        build_timestamp: env!("VERGEN_BUILD_TIMESTAMP").to_string(),
+    }
+}
+
+/// Global static version metadata
+static VERSION_METADATA: OnceLock<RethCliVersionConsts<'static>> = OnceLock::new();
+
+/// Initialize the global version metadata.
+pub fn init_version_metadata(metadata: RethCliVersionConsts<'static>) {
+    let _ = VERSION_METADATA.set(metadata);
+}
+
+/// Get a reference to the initialized version metadata.
+///
+/// # Panics
+/// If init_version_metadata() hasn't been called.
+pub fn version_metadata() -> &'static RethCliVersionConsts<'static> {
+    VERSION_METADATA.get().expect("Version metadata not initialized")
```

Comment:
for this we want a `get_version_metadata` that does an or_init with the `default_version_metadata` so that this function does the init